test/e2e/multihoming.go

```diff
@@ -2198,53 +2524,524 @@ func computeIPWithOffset(baseAddr string, increment int) (string, error) {
 	return netip.PrefixFrom(ip, addr.Bits()).String(), nil
 }
 
-// Given a node name and an offset, generateIPsFromNodePrimaryIfAddr returns an IPv4 and an IPv6 address
-// at the provided offset from the primary interface addresses found on the node.
-func generateIPsFromNodePrimaryIfAddr(cs clientset.Interface, nodeName string, offset int) ([]string, error) {
-	var newAddresses []string
-
-	node, err := cs.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+// buildRouteToHostSubnetViaExternalContainer returns ip route add commands to reach the host subnets via the provided gateway IPs
+func buildRouteToHostSubnetViaExternalContainer(cs clientset.Interface, f *framework.Framework, nodeName string, gwV4, gwV6, interfaceName string) ([]string, error) {
+	cmdTemplateV4 := "ip -4 route replace %s via " + gwV4 + " dev " + interfaceName
+	cmdTemplateV6 := "ip -6 route replace %s via " + gwV6 + " dev " + interfaceName
+	cmds := []string{}
+	hostSubnets, err := getHostSubnetsForNode(cs, nodeName)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to get node %s: %v", nodeName, err)
+		return nil, fmt.Errorf("failed to get host subnets for node %s: %w", nodeName, err)
 	}
+	for _, hostSubnet := range hostSubnets {
+		if utilnet.IsIPv4CIDRString(hostSubnet) && gwV4 != "" {
+			cmds = append(cmds, fmt.Sprintf(cmdTemplateV4, hostSubnet))
+		} else if utilnet.IsIPv6CIDRString(hostSubnet) && gwV6 != "" {
+			cmds = append(cmds, fmt.Sprintf(cmdTemplateV6, hostSubnet))
+		}
+	}
+	return cmds, nil
+}
 
-	nodeIfAddr, err := util.GetNodeIfAddrAnnotation(node)
+// Given a node name and an offset, generateIPsFromNodePrimaryNetworkAddresses returns an IPv4 and an IPv6 address
+// at the provided offset from the primary interface network addresses found on the node.
+func generateIPsFromNodePrimaryNetworkAddresses(cs clientset.Interface, nodeName string, offset int) ([]string, error) {
+	hostSubnets, err := getHostSubnetsForNode(cs, nodeName)
 	if err != nil {
 		return nil, err
 	}
-	nodeAddresses := []string{}
-	if nodeIfAddr.IPv4 != "" {
-		nodeAddresses = append(nodeAddresses, nodeIfAddr.IPv4)
+	return generateIPsFromSubnets(hostSubnets, offset)
+}
+
+func addIPRequestToPodConfig(cs clientset.Interface, podConfig *podConfiguration, offset int) error {
+	nodeName, ok := podConfig.nodeSelector[nodeHostnameKey]
+	if !ok {
+		return fmt.Errorf("no node selector found on podConfig")
 	}
-	if nodeIfAddr.IPv6 != "" {
-		nodeAddresses = append(nodeAddresses, nodeIfAddr.IPv6)
+
+	var (
+		IPsToRequest []string
```

Comment:
pedantic nit: not sure this counts as breaking the variable naming convention (start w/ lowercase). 

```suggestion
		ipsToRequest []string
```